docs/testing/unit-tests.md

```diff
@@ -1,3 +1,247 @@
 # Unit Tests
 
-TBD
+As well fostering code quality, well-written unit tests provide a fine way for developers to quickly understand how to run a piece of code, and what it does under various conditions. It's important therefore to maintain high standards not just in our source code, but in our tests as well. 
+
+This guide is intended as a quick reference of common tools and conventions we use for Calypso unit testing. You'll be able to glean a great deal from browsing through Calypso's existing tests, but we'll try to keep this page up to date with the most recent standards.
+
+## Table of contents
+* [Running tests](running-tests)
+* [Writing tests](writing-tests)
+    * [Folder structure](#folder-structure)
+    * [Describing tests](#describing-tests)
+    * [Snapshot testing](#snapshot-testing)
+* [Examples](#examples)
+
+## Running tests
+
+See [testing-overview.md](testing-overview.md).
+
+## Writing tests
+
+We use [Jest](https://facebook.github.io/jest) testing framework for unit tests, and [Enzyme](https://github.com/airbnb/enzyme) for testing React components. You can find more information on testing components at [component tests](component-tests.md)._
+
+Though you'll still see Chai assertions and Sinon spies/mocks throughout the code base, for all new tests we use the Jest API for [globals](https://facebook.github.io/jest/docs/en/mock-function-api.html#content) (`describe`, `test`, `beforeEach` and so on) [assertions](http://facebook.github.io/jest/docs/en/expect.html#content), [mocks](http://facebook.github.io/jest/docs/en/mock-functions.html#content) and [mock function/spies](https://facebook.github.io/jest/docs/en/mock-function-api.html#content).
+
+## Folder structure
+Keep your tests in a `test` folder in your working directory. The test file should have the same name as the test subject file, and use the extension `.js`.
+
```

Comment:
Can we also make it explicit that in the `test` folder can be placed only files that contain at least one test? Otherwise, Jest will load the file and throw an error. That's why all fixtures or test helpers should be put in subfolders.

Reply:
Do we want to talk about how to import tested code? There seems to be a tendency for relative imports, `import { foo } from '../foo';` which is nice because the tests can move around with the source. Occasionally I see project base paths imports in tests (`import { thingToTest } from 'thing/totest'`). I have a slight preference for relative imports of the _tested_ code in an immediate parent.

Reply:
> Can we also make it explicit that in the test folder can be placed only files that contain at least one test?

Yep, good idea. I don't want to duplicate the info in _testing-overview.md_ so I'll let this doc subsume it. :)

> Do we want to talk about how to import tested code? There seems to be a tendency for relative imports, import { foo } from '../foo';

Makes sense to include a recommendation to this effect 👍

Reply:
Why is the relative import preferred? I find that relative imports in general make it harder to analyze code in codemods because the paths get removed. Even if we move the folder (which happens rarely) our tests would immediately fail and we should be able to see the cause (broken import path).

Reply:
I'm open to discussion.

Here are a few advantages I see with relative imports:

* Test directory is relative to tested code, import reflects this
* Test directory moves with parent in case that happens
* Isolated from changes to module paths (is this `server`, `client`, `extension`, or `node_modules` based import?)